File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators = { "*", "+", "/", "-" }
        s = []

        for t in tokens:
            res = 0
            if t not in operators:
                res = int(t)
            elif t == "*":
                res = s.pop() * s.pop()
            elif t == "+":
                res = s.pop() + s.pop()
            elif t == "-":
                res = -(s.pop())
                res += s.pop()
            else:
                op = s.pop()
                res = int(s.pop() / op)
            s.append(res)
            
        return s.pop()

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Set<String> operators = new HashSet<>(Arrays.asList("+", "-", "*", "/"));
        Stack<Integer> s = new Stack<>();

        for (String t : tokens) {
            if (!operators.contains(t)) {
                s.push(Integer.valueOf(t));
            }
            else {
                int res = 0;
                if (t.equals("+")) {
                    res = s.pop() + s.pop();
                }
                else if (t.equals("-")) {
                    res += -(s.pop());
                    res += s.pop();
                }
                else if (t.equals("*")) {   
                    res = s.pop() * s.pop();
                }
                else {
                    int op;
                    op = s.pop();
                    res = s.pop() / op;
                }
                s.push(res);
            }
        }

        return s.pop();
    }
}